src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -100,13 +104,14 @@ public User create(User user) {
     @Override
     public User update(User user) {
         String query = "UPDATE users  "
-                + "SET user_name = ?, login = ?, user_password = ? "
+                + "SET user_name = ?, login = ?, user_password = ?, salt = ?"
                 + "WHERE user_id = ? AND user_deleted = FALSE;";
         try (Connection connection = ConnectionUtil.getConnection()) {
             try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
                 preparedStatement.setString(1, user.getName());
                 preparedStatement.setString(2, user.getLogin());
                 preparedStatement.setString(3, user.getPassword());
+                preparedStatement.setBytes(4, user.getSalt());
                 preparedStatement.executeUpdate();
```

Comment:
how many arguments `?` are there in your query? should they all be set for the prepared statement to work correctly?

Reply:
added 5-th set for id